ios/ZulipMobile.xcodeproj/project.pbxproj

```diff
@@ -175,7 +175,7 @@
 		83CBB9F71A601CBA00E9B192 /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 940;
+				LastUpgradeCheck = 1160;
```

Comment:
Thanks for catching this!!

Reply:
Er, hmm, actually when I go through that process, that diff hunk doesn't get applied. Looking into this now.

Reply:
It seems to relate to this, from the [Xcode 10.2 release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-10_2-release-notes):

> Xcode now more carefully distinguishes between legacy localization identifiers such as “English” and modern localization identifiers such as “en” and represents them both in both project files and the user interface. (45469882)

I found that from a CocoaPods issue, https://github.com/CocoaPods/Xcodeproj/pull/669.

The specific deprecation warning, according to that issue, is this:

> Migrating the English, deprecated localization to English is recommended for all projects. This will ensure localized resources are placed in “en.lproj” directories instead of deprecated “English.lproj” directories.

That sure sounds familiar, but I can't seem to get the same warning.

Reply:
Aha! So, starting from a point where `LastUpgradeVersion` and `LastUpgradeCheck` are 0940 in the three places in the commit in question, then clearing the build folder and quitting and restarting Xcode, I do get that warning:

![Screen Shot 2020-09-04 at 3 11 44 PM](https://user-images.githubusercontent.com/22248748/92288315-69d6b900-eec1-11ea-8ed2-8e439cdcade3.png)

and the fix does add that hunk.

...Then, it seems Xcode has forgotten about the other warning that I think gave us the `CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED` setting...In the [upgrade guide](https://github.com/react-native-community/upgrade-support/issues/13#issue-589114534), that looks like this:

![image](https://user-images.githubusercontent.com/22248748/92288502-f2edf000-eec1-11ea-8f8a-ab6d3977188d.png)

In general it's been frustrating trying to see these deprecation warnings (and be able to fix them automatically!) in cases where you'd think they'd apply. Anyway, it seems like we've pretty much nailed down the changes in https://github.com/facebook/react-native/commit/ebb629d05 as things we want to take in this commit, so maybe I'll go ahead and do that.

Reply:
Great!

Does this complete #4112, or is there more to that?

Reply:
Well, with some of these notices seemingly randomly absent at times when they should show up, it's hard to be sure. There are also several of these that seem to apply to our dependencies. One that happens to show up now (that I don't remember seeing except recently, maybe yesterday?) is listed under our ZulipMobile workspace but looks like it's about Yoga, which is a React Native thing:

![image](https://user-images.githubusercontent.com/22248748/92962473-d1db5100-f425-11ea-8be7-d340fca6ca54.png)

And there's always been a handful under the Pods project:

![image](https://user-images.githubusercontent.com/22248748/92962515-e3bcf400-f425-11ea-8de4-f80444c5c144.png)

I don't think there's a strong reason we should touch anything to do with our dependencies; morally, their maintainers should handle them, but also, I suspect anything we do to them might get clobbered by `pod install`.

There's a handful of deprecation complaints about our native code itself (as opposed to various settings in Xcode) that seem to center on notifications:

![image](https://user-images.githubusercontent.com/22248748/92963111-cb99a480-f426-11ea-85f5-c20bf8dac47c.png)

But I suspect most of these will go away with https://github.com/zulip/zulip-mobile/pull/4163. If I remember correctly, a few will remain, but they'll mostly appear alongside comments in the code that acknowledge a deprecation and say that we're using the deprecated API on purpose, for compatibility with older iOS versions.